src/main.py

```diff
@@ -40,15 +46,23 @@ def setup(self):
         for sprite in self.hud.hud_sprite_list:
             self.all_sprites.append(sprite)
 
-    def on_draw(self):
-        # clear screen
-        self.clear()
-        self.all_sprites.draw()
-        for player in self.player_manager.players:
-            player.draw()
-        self.input_debug_hud.draw()
-
     def on_update(self, delta_time):
+        # In the debugger, we intentionally ignore the on_update() call we get from arcade engine.
+        # Instead, we call our own our_update() within on_draw()
+        # This avoids a known issue where debugger pauses can cause `on_update`` and `on_draw`` to happen in this order:
+        #   update
+        #   update
+        #   draw
+        #   draw
+        #   update
+        #   update
+        #   draw
+        #   draw
+        #   ...and so on.  Each is called twice, back-to-back.
```

Comment:
This big comment block was helpful for figuring out what you did, but is it meant to live here?

Reply:
I was gonna leave it yeah.  But I can move it into `constants.py` if you prefer?